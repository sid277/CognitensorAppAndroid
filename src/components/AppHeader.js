import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { theme } from '../theme';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { TouchableOpacity } from 'react-native-gesture-handler';

const AppHeader = ({ scene, previous, navigation, searchVisible = false }) => {
  const { options } = scene.descriptor;
  const title =
    options.headerTitle !== undefined
      ? options.headerTitle
      : options.title !== undefined
      ? options.title
      : scene.route.name;
  const dropShadowStyle = searchVisible ? styles.dropShadow : {};

  return (
    <View style={styles.header}>
      <View style={styles.headerLeftIcon}>
        <TouchableOpacity onPress={navigation.pop}>
          {previous ? (
            <MaterialIcons name="chevron-left" size={24} style={styles.visible} />
          ) : (
            <MaterialIcons
              name="chevron-left"
              size={24}
              style={styles.invisible}
            />
          )}
        </TouchableOpacity>
      </View>
      <Text style={styles.headerText}>{title}</Text>
      <View style={[styles.headerRightIcon, dropShadowStyle]}>
        {searchVisible ? (
          <MaterialIcons name="search" size={24} style={styles.visible} />
        ) : (
          <MaterialIcons name="search" size={24} style={styles.invisible} />
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    backgroundColor: '#FFF',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: 60,
    flexDirection: 'row',
    padding: theme.spacing.small,
  },
  headerText: {
    ...theme.typography.title2,
    fontWeight: '700',
  },
  headerLeftIcon: {
    marginRight: 'auto',
    padding: theme.spacing.tiny,
  },
  headerRightIcon: {
    marginLeft: 'auto',
    padding: theme.spacing.tiny,
  },
  invisible: {
    color: theme.colors.appbar,
  },
  visible: {
    color: '#000',
  },
  dropShadow: {
    borderRadius: 50,
    backgroundColor: 'white',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.2,
    shadowRadius: 1.41,
    elevation: 2,
  },
});

export default AppHeader;
