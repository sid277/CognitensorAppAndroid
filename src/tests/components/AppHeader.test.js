import React from 'react';
import { shallow } from 'enzyme';
import AppHeader from '../../components/AppHeader';
import { create } from 'react-test-renderer';


test('should test AppHeader component', () => {
    const scene = {
        descriptor : {
            options : {
                title : "Title_test1",
        }},
      }
      const previous = "previous";
      const navigation = {
        pop : "test_pop",
      }

const wrapper = create(<AppHeader scene={scene} navigation ={navigation}/>);
expect(wrapper).toMatchSnapshot();
});

// test('renders correctly', () => {
//     const tree = renderer.create(<AppHeader />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });