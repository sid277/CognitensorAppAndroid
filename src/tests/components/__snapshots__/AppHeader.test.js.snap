// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test AppHeader component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "center",
      "padding": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": "auto",
        "padding": 8,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              Object {
                "color": "#ffffff",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "700",
      }
    }
  >
    Title_test1
  </Text>
  <View
    style={
      Array [
        Object {
          "marginLeft": "auto",
          "padding": 8,
        },
        Object {},
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 24,
          },
          Object {
            "color": "#ffffff",
          },
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
